### PR TITLE
355 customize template

### PR DESCRIPTION
**Pull request by kumaravel29**
To support dynamic template support, 2 new settings customize_template and index_prefix have been added. With this change, the users can use a dynamic template containing some identifiers which would be mentioned in the configuration and can be replaced during the boot of the plugin.
NOTE: The above changes have been tested using the plugin in combination with the forest plugin

README updated
README Table of Contents updated
History.md and version in gemspec are untouched

### CODE REVIEW

**Comment by kumaravel29** [comment]
I am not sure how to solve this conflict

File: README.md [hunk 1]
```diff
@@ -35,8 +35,9 @@ Current maintainers: @cosmo0920
   + [template_name](#template_name)
   + [template_file](#template_file)
   + [template_overwrite](#template_overwrite)
+  + [customize_template](#customize_template)
+  + [index_prefix](#index_prefix)
   + [templates](#templates)
-  + [max_retry_putting_template](#max_retry_putting_template)
```
Comment (cosmo0920) [inline README.md]: I don’t want to remove this line.

Comment (kumaravel29) [reply README.md]: I have added the line again.

**Comment by kumaravel29** [comment]
I don't have any changes to add testcases for dynamic templates.

**Comment by cosmo0920** [comment]
Could you add testcases for dynamic template feature?

**Comment by kumaravel29** [comment]
@cosmo0920 I have added 2 test cases for the custom/dynamic template.

**Pull request merged**